packages/flutter_tools/lib/runner.dart

```diff
@@ -128,18 +132,52 @@ Future<int> _handleToolError(
       command: args.join(' '),
     );
 
-    if (error is String) {
-      stderr.writeln('Oops; flutter has exited unexpectedly: "$error".');
-    } else {
-      stderr.writeln('Oops; flutter has exited unexpectedly.');
-    }
+    final String command = _crashCommand(args);
```

Comment:
Consider whether moving some or all of the code under the else on 115 into a separate method would improve readability of `_handleToolError`.

Reply:
There was enough complicated try/catch and early returns that I just pulled out the wall-o-text part.  Hopefully that makes it readable enough.

Reply:
I moved it into its own file.